arrays/arrays.c

```diff
@@ -34,10 +38,13 @@ Array *create_array (int capacity) {
  *****/
 void destroy_array(Array *arr) {
 
-  // Free all elements
-
-  // Free array
-
+      // Free all elements-- first
+      for(int i = 0; i < arr->count; i++) {
+            arr->elements[i] = NULL;
+            free(arr->elements[i]);
+       }
+      // Free array -- second
+      free(arr);
 }
 
```

Comment:
Everything looks good so far! As far as line 43 goes, I don't believe you need that there.

Reply:
yes, forgot to comment it.